services/feeds.js

```diff
@@ -0,0 +1,160 @@
+var request = require('superagent');
+
+/**
+ * @constructor
+ */
+
+function Feeds(defaults) {
+
+	defaults = defaults || {};
+
+	/* set default format or lang for all requests */
+
+	/* we default to json since it's a js-sdk */
+	this.format = defaults.format || 'json';
+	this.lang = defaults.lang;
+}
+
+/**
+ * Creates an object with the option feed params
+ * for this request.
+ * @returns {Object}
+ */
+
+Feeds.prototype.params = function (args) {
+	var params;
+
+	args = args || {};
+
+	params = {
+		format: args.format || this.format
+	};
+
+	/* don't include unless explicitly set */
+	if (args.lang || this.lang) {
+		params.lang = args.lang || this.lang;
+	}
+	return params;
+};
+
+/**
+ * Asserts that `key` is present in `args`.
+ * @param {Object} args
+ * @param {String} key
+ * @throws {Error}
+ */
+
+function assert(args, key) {
+	if (!args || !args.hasOwnProperty(key)) {
+		throw new Error('Missing required argument "' + key + '"');
+	}
+}
+
+/**
+ * Public Photos & Video
+ * @see https://www.flickr.com/services/feeds/docs/photos_public/
+ */
+
+Feeds.prototype.publicPhotos = function (args) {
+	return request('GET', 'https://www.flickr.com/services/feeds/photos_public.gne')
```

Comment:
It looks like generally there’s a “photos_public.gne” → “publicPhotos” reversal happening here. It might be a little clearer about what connects to what if we didn't reverse. So `photosPublic`, `photosFriends`, `photosFaves`, etc.